pennylane/plugins/default_qubit.py

```diff
@@ -213,7 +221,42 @@ def mat_vec_product(self, mat, vec, wires):
         perm = wires + unused_idxs
         inv_perm = np.argsort(perm)  # argsort gives inverse permutation
         state_multi_index = np.transpose(tdot, inv_perm)
-        return np.reshape(state_multi_index, 2 ** self.num_wires)
+
+        return state_multi_index
+
+    def mat_vec_product_einsum(self, mat, vec, wires):
```

Comment:
The integration tests are picking this part of the code nice, but it could be nice to include a unit test for it as well.